File: LevkoAndPermutation.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;

int main(){
    int n, k;
    cin >> n >> k;
    bool flag = 1;
    if (k > n-1){
        cout << -1 << "\n";
    }
    else{
        int current=0, last = n-1, operations = n - k -1;
        vector<int> numbers;
        for (int i=1; i<=n; i++){
            numbers.push_back(i);
        }
        while (operations > 0){
            operations -= 1;
            swap(numbers[current], numbers[last]);
            last -= 1;
        }
        for (int j : numbers){
            cout << j << " ";
        }
    }
}

File: LevkoAndPermutation.py
from operator import truediv


n, k = map(int, input().split())
gate = True
if k > n-1:
    gate = False

operations = n - k -1
current, last = 0, n-1
numbers = list(range(1, n+1))

while operations > 0 and k < n-1 and gate:
    operations -= 1
    numbers[current], numbers[last] = numbers[last], numbers[current]
    last -= 1

print(*numbers) if gate else print(-1)
